### PR TITLE
handle windows byte order mark, ran into this on adal

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: depfinder/main.py [hunk 1]
```diff
@@ -260,7 +260,7 @@ def parse_file(python_file):
         PACKAGE_NAME = os.path.basename(python_file).split('.')[0]
         logger.debug("Setting PACKAGE_NAME global variable to {}"
                      "".format(PACKAGE_NAME))
-    with open(python_file, 'r') as f:
+    with open(python_file, 'r', encoding='utf-8-sig') as f:
```
Comment (ericdill) [inline depfinder/main.py:263]: Is doing this for _all_ files a good idea?

Comment (CJ-Wright) [reply depfinder/main.py:263]: It seemed safe for the files I tried to run it on, but I'm not certain.

Comment (ericdill) [reply depfinder/main.py:263]: I'd be interested in two possible paths here. One hacky and one less so. The hacky way would be to just catch the exception that's being raised by the adal code, set the encoding and open it again. THe other way would be to use something like python-magic or chardet to attempt to automatically detect the encoding and use it to open the file. I'd be fine with just a try/except to fix this adal problem

Comment (CJ-Wright) [reply depfinder/main.py]: I've pushed up the try except and it worked on my adal test

**Comment by CJ-Wright** [comment]
@ericdill ready for another round!

File: depfinder/main.py [hunk 2]
```diff
@@ -260,9 +260,14 @@ def parse_file(python_file):
...
                      "".format(PACKAGE_NAME))
-    with open(python_file, 'r') as f:
-        code = f.read()
-    catcher = get_imported_libs(code)
+    try:
+        with open(python_file, 'r') as f:
+            code = f.read()
+        catcher = get_imported_libs(code)
+    except SyntaxError:
+        with open(python_file, 'r', encoding='utf-8-sig') as f:
```
Comment (ericdill) [inline depfinder/main.py:268]: Can you add an inline note here that this is specifically to catch a problem discovered when analyzing the adal source? After that go ahead and merge and cut another release if you need one

**Pull request merged**